tests/test_alpm.py

```diff
@@ -0,0 +1,110 @@
+# MIT licensed
+# Copyright (c) 2020 DDoSolitary <DDoSolitary@gmail.com>, et al.
+
+import pytest
+import os
+import shutil
+import subprocess
+import tempfile
+
+pytestmark = [
+  pytest.mark.asyncio,
+  pytest.mark.skipif(shutil.which('makepkg') is None, reason='requires makepkg command'),
+  pytest.mark.skipif(shutil.which('repo-add') is None, reason='requires repo-add command')
+]
+
+global temp_dir, pkg_dir, db_dir
+
+
+def setup_module(module):
+  global temp_dir, pkg_dir, db_dir
+  temp_dir = tempfile.TemporaryDirectory()
+  pkg_dir = os.path.join(temp_dir.name, 'test-pkg')
+  os.mkdir(pkg_dir)
+  with open(os.path.join(pkg_dir, 'PKGBUILD'), 'w') as f:
+    f.write(
+      'pkgname=test-pkg\n'
+      'pkgver=1.2.3\n'
+      'pkgrel=4\n'
+      'arch=(any)\n'
+      'provides=("test-provides=5.6-7" "test-provides-unversioned")\n'
+    )
+  subprocess.check_call(['makepkg'], cwd=pkg_dir)
+  pkg_file = subprocess.check_output(['makepkg', '--packagelist'], cwd=pkg_dir, text=True).strip()
+  db_dir = os.path.join(temp_dir.name, 'test-db')
+  os.mkdir(db_dir)
+  repo_dir = os.path.join(db_dir, 'sync')
+  os.mkdir(repo_dir)
+  subprocess.check_call([
+    'repo-add',
+    os.path.join(repo_dir, 'test-repo.db.tar.gz'),
+    os.path.join(pkg_dir, pkg_file)
+  ])
```

Comment:
Too many `os.path`s. Try to use `pathlib`?

Reply:
Done.